### ISSUE TITLE
Timeout issue

### ISSUE THREAD

**Issue by bhavikpatel7023** [bitshiftza/fnb-api#2]

**Comment by DanielHolland** [comment]
@bhavikpatel7023 you might want to obscure the credentials in your image, it's possible to log into the online banking profile using the credentials in the screenshot

**Comment by AntonJanHart** [comment]
Would like to know if you have made any progress with timeout issue, sitting with the same problem:

```
> fnb-api@1.0.9 test C:\wamp64\www\fnb-api-master
> ts-node --project tsconfig.test.json ./test/index.ts

(node:6168) UnhandledPromiseRejectionWarning: TimeoutError: waiting for function failed: timeout 30000ms exceeded
    at new WaitTask (C:\wamp64\www\fnb-api-master\node_modules\puppeteer\lib\DOMWorld.js:549:28)
    at DOMWorld.waitForFunction (C:\wamp64\www\fnb-api-master\node_modules\puppeteer\lib\DOMWorld.js:454:12)
    at Frame.waitForFunction (C:\wamp64\www\fnb-api-master\node_modules\puppeteer\lib\FrameManager.js:657:28)
    at Page.waitForFunction (C:\wamp64\www\fnb-api-master\node_modules\puppeteer\lib\Page.js:1141:29)
    at Scraper.<anonymous> (C:\wamp64\www\fnb-api-master\src\scrapers\scraper.ts:217:57)
    at step (C:\wamp64\www\fnb-api-master\src\scrapers\scraper.ts:33:23)
    at Object.next (C:\wamp64\www\fnb-api-master\src\scrapers\scraper.ts:14:53)
    at fulfilled (C:\wamp64\www\fnb-api-master\src\scrapers\scraper.ts:5:58)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
(node:6168) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by reje
cting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://
nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:6168) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process
with a non-zero exit code.
Terminate batch job (Y/N)? y
```

**Comment by AntonJanHart** [comment]
Hi guys.
Occasionally still recieving timeout errors, mostly ons every second execution:
`TimeoutError: waiting for function failed: timeout 30000ms exceeded`
Question, does pupeteer do a logout once done scraping?
All I can think of is that fnb does not get logged out properly befor the next run.
Regards

**Comment by MorneBrummer** [comment]
Hi Devs, 

Any luck on the timeout issue ? 

Regards

**Comment by circlingthesun** [comment]
I've found that a security warning would occasionally pop up on login and make the scraping timeout, I've submitted a PR here: https://github.com/bitshiftza/fnb-api/pull/16

**Comment by bitshiftza** [comment]
@AntonJanHart @MorneBrummer  @LukeV111 @circlingthesun just published a new version (1.0.11) with the timeout changes from @circlingthesun.

I've noticed that it'll also throw a timeout if the password is incorrect but I'll address that in a separate issue. Going to close this, if you get any other timeouts, feel free to open another one
